app/models/event.rb

```diff
@@ -66,6 +66,17 @@ def search_activities(author = nil, query_string = "", filter = "all")
   end
   alias_method :activities, :search_activities
 
+  def activities_per_day(author = nil, query_string = "", filter = "all")
+    activities = search_activities(author, query_string, filter)
+    activities.sort_by! { |activity| activity.dates.first }
+
+    activities.each_with_object(Hash.new { [] }) do |activity, grouped_by_day|
```

Comment:
I think this is a block that returns the default entry if the key is not found. And in this case the default is a new array for every lookup that has no key. That way line 75 is possible.